### PR TITLE
Starting expansion of ZKVM rustdocs

### PR DESCRIPTION
**Pull request by 3lkn**

### CODE REVIEW

**Review by tzerrell — CHANGES_REQUESTED** [review]
`cargo doc` is currently broken and this probably shouldn't land without it running. That said, the problem seems to be on `main` and not this PR -- we should add `cargo doc` or a variant to CI, either in this PR or a separate one, to catch this in the future.

File: risc0/zkvm/sdk/rust/src/receipt.rs
Comment (tzerrell) [inline risc0/zkvm/sdk/rust/src/receipt.rs:103]: IIUC,  this is the generalized verify, which allows the caller to select whatever HAL they like (whereas `verify` uses the default SHA & CPU HAL). The way this comment is framed feels a bit backwards to me -- it doesn't exist to be called by `verify`, `verify` exists to provide default argument to `verify_with_hal`.

**Review by tzerrell — APPROVED** [review]
cool!

File: risc0/zkvm/sdk/rust/guest/src/env.rs
Comment (flaub) [inline risc0/zkvm/sdk/rust/guest/src/env.rs:113]:
```suggestion
/// Values are read in the order in which they are written by the host,
```

File: risc0/zkvm/sdk/rust/guest/src/env.rs
Comment (flaub) [inline risc0/zkvm/sdk/rust/guest/src/env.rs:115]: On the host side, the call would be `prover.add_input` or some such.

File: risc0/zkvm/sdk/rust/guest/src/env.rs
Comment (flaub) [inline risc0/zkvm/sdk/rust/guest/src/env.rs:122]: It's be better if we had a self-contained example so that:
1. it requires less context to understand the idea
2. it doesn't depend on another repository where the link might eventually fail

File: risc0/zkvm/sdk/rust/guest/src/env.rs
Comment (flaub) [inline risc0/zkvm/sdk/rust/guest/src/env.rs:138]: The host would receive this value thru accessing the journal, `env` is only used on the guest-side.

File: risc0/zkvm/sdk/rust/guest/src/env.rs
Comment (flaub) [inline risc0/zkvm/sdk/rust/guest/src/env.rs:188]:
```suggestion
/// let cycles_elapsed = count2 - count1;
```

File: risc0/zkvm/sdk/rust/src/receipt.rs
Comment (flaub) [inline risc0/zkvm/sdk/rust/src/receipt.rs:107]: Actually the HAL is independent of the circuit, it's the "Hardware Abstraction Layer" which provides a way to execute portions of the verification in an efficient way based on whether the verify is happening on the host or in the guest (aka recursion).

File: risc0/zkvm/sdk/rust/guest/src/env.rs
Comment (flaub) [inline risc0/zkvm/sdk/rust/guest/src/env.rs:99]: Maybe word this like:
"See `env::write()` for passing structured data to the host".

Basically it's not really preferred, it just depends on what the needs of the user are. I have a feeling we're going to pretty significantly revamp this part of the API once the zero-copy deserialize lands. There's already some open PRs to provide a better API to allow raw-byte access (which I think will become obsolete when zero-copy lands).

File: risc0/zkvm/sdk/rust/guest/src/env.rs
Comment (flaub) [inline risc0/zkvm/sdk/rust/guest/src/env.rs:37]:
```suggestion
/// This is a stub version of `risc0_zkvm_platform::rt::host_sendrecv`,
```

Or perhaps make an actual rustdoc link here.

File: risc0/zkvm/sdk/rust/guest/src/sha.rs
Comment (flaub) [inline risc0/zkvm/sdk/rust/guest/src/sha.rs:90]:
```suggestion
// marker and trailer, in order to take the SHA hash of `len_bytes` bytes.
```

File: risc0/zkvm/sdk/rust/build/src/lib.rs
Comment (pdg744) [inline risc0/zkvm/sdk/rust/build/src/lib.rs:411]: perhaps: 

`... increasing code_limit by 1...`

File: risc0/zkvm/sdk/rust/build/src/lib.rs
Comment (pdg744) [inline risc0/zkvm/sdk/rust/build/src/lib.rs:406]: Perhaps just:

/// The number of <s>po2</s> entries <s>to generate</s> in the MethodID.

File: risc0/zkvm/sdk/rust/guest/src/env.rs
Comment (pdg744) [inline risc0/zkvm/sdk/rust/guest/src/env.rs:182]: rad

File: risc0/zkvm/sdk/rust/src/lib.rs
Comment (pdg744) [inline risc0/zkvm/sdk/rust/src/lib.rs:27]: nice

File: risc0/zkvm/sdk/rust/src/receipt.rs
Comment (pdg744) [inline risc0/zkvm/sdk/rust/src/receipt.rs:72]: Perhaps: 

`Verifies a RISC Zero Receipt`

**Review by tzerrell — APPROVED** [review]
LGTM once CI is happy

**Pull request merged**
